src/Chapter1.hs

```diff
@@ -519,9 +521,16 @@ branches because it is an expression and it must always return some value.
 👩‍🔬 Due to lazy evaluation in Haskell, only the expression from the branch
   satisfying the check will be returned and, therefore, evaluated.
 -}
-closestToZero :: Int -> Int -> Int
-closestToZero x y = error "closestToZero: not implemented!"
 
+closestToZero :: Integral a => a -> a -> a
+closestToZero x y 
+  | x == 0 || y == 0 = 0
+  | x == y    = x
+  | x' > y'   = y
+  | x' < y'   = x
+  | x' == y'  = if x > y then x else y
+  where x' = abs (x - 0)
+        y' = abs (y - 0)
```

Comment:
That's interesting solution! 
For this particular problem, what we only need to know to give the answer is: `abs`olute value of which of `x` or `y` is smaller. Being smaller would make it closer to zero, so basically our check could look like:

```haskell
closestToZero x y = if abs x < abs y then x else y
```